torch/_dynamo/variables/builtin.py

```diff
@@ -498,11 +485,45 @@ def call_range(self, tx, *args, **kwargs):
                     **{k: v.value for k, v in kwargs.items()},
                 ),
             )
+        elif self._dynamic_args(*args, **kwargs):
+            assert len(kwargs) == 0
+
+            def guard_if_dyn(arg):
+                if isinstance(arg, DynamicShapeVariable):
+                    return arg.evaluate_expr(tx.output)
+                return arg
+
+            args = [guard_if_dyn(arg) for arg in args]
+            value = self.fn(*args)
+            return variables.RangeVariable(
+                value=value
+            )
+        # None no-ops this handler and lets the driving function proceed
+        return None
 
     def call_slice(self, tx, *args):
         return variables.SliceVariable(args)
 
-    def _call_iter_tuple_list(self, tx, obj=None):
+    def _dyn_proxy(self, tx, *args, **kwargs):
+        assert self._dynamic_args(*args, **kwargs)
+        from .builder import wrap_fx_proxy
+
+        options = VariableTracker.propagate(self, args, kwargs.values())
+        return wrap_fx_proxy(
+            tx,
+            tx.output.create_proxy(
+                "call_function", self.fn, *proxy_args_kwargs(args, kwargs)
+            ),
+            **options,
+        )
+
+    def call_mod(self, tx, *args, **kwargs):
+        if self._dynamic_args(*args, **kwargs):
+            return self._dyn_proxy(tx, *args, **kwargs)
+
+    def _call_iter_tuple_list(self, tx, obj=None, *args, **kwargs):
+        if self._dynamic_args(*args, **kwargs):
+            return self._dyn_proxy(tx, *args, **kwargs)
```

Comment:
I'm not sure why iter tuple list gets the same dynamic shapes behavior as mod...